### PR TITLE
Simplified BufferEncoder/Decoder APIs

### PR DESCRIPTION
**Pull request by KrzysztofCwalina**
Simplified some APIs based on a discussion with @tarekgh.

cc: @ahsonkhan, @joshfree, @shiftylogic

### CODE REVIEW

File: src/System.Text.Encodings.Web.Utf8/UrlDecoder.cs [hunk 1]
```diff
@@ -2,7 +2,11 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System.Text.Encodings.Web.Buffers;
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
```
Comment (tarekgh) [inline src/System.Text.Encodings.Web.Utf8/UrlDecoder.cs]: remove it please

File: src/System.Azure.Experimental/System/Azure/Key.cs [hunk 2]
```diff
@@ -13,14 +13,15 @@ namespace System.Azure.Authentication
     public static class Key {
         public static byte[] ComputeKeyBytes(string key)
         {
-            const int bufferLength = 128;
+            int size = key.Length * 2;
+            var buffer = size < 128 ? stackalloc byte[size] : (Span<byte>)new byte[size];
```
Comment (ahsonkhan) [inline src/System.Azure.Experimental/System/Azure/Key.cs]: nit: change var to Span<byte>. Removes the need for explicit cast.
```C#
Span<byte> buffer = size < 128 ? stackalloc byte[size] : new byte[size]; 
```

Comment (KrzysztofCwalina) [reply src/System.Azure.Experimental/System/Azure/Key.cs]: Does not work today. @VSadov started a discussion  to enable it.

File: src/System.Text.Encodings.Web.Utf8/UrlDecoder.cs (hunk 1 above)
Comment (KrzysztofCwalina) [reply src/System.Text.Encodings.Web.Utf8/UrlDecoder.cs]: This is a standard license we use in corfxlab, isn't it?

File: src/System.Azure.Experimental/System/Azure/Key.cs (hunk 2 above)
Comment (ahsonkhan) [reply src/System.Azure.Experimental/System/Azure/Key.cs]: Ah yes. I forgot:
https://github.com/dotnet/corefxlab/pull/1742/files#r137580133

File: src/System.Text.Encodings.Web.Utf8/UrlDecoder.cs (hunk 1 above)
Comment (ahsonkhan) [reply src/System.Text.Encodings.Web.Utf8/UrlDecoder.cs]: Its duplicated.

**Comment by KrzysztofCwalina** [comment]
@dotnet-bot test this

**Comment by KrzysztofCwalina** [comment]
@dotnet-bot test Innerloop Windows_NT Release Build and Test

**Comment by KrzysztofCwalina** [comment]
@dotnet-bot test Innerloop

**Comment by KrzysztofCwalina** [comment]
@dotnet-bot test this

**Comment by KrzysztofCwalina** [comment]
@dotnet-bot test Innerloop OSX10.12 Debug Build and Test

**Comment by KrzysztofCwalina** [comment]
@dotnet-bot  Innerloop Windows_NT Release Build and Test

**Comment by KrzysztofCwalina** [comment]
@dotnet-bot test Innerloop OSX10.12 Release Build and Test

**Comment by KrzysztofCwalina** [comment]
@dotnet-bot Innerloop Windows_NT Debug Build and Test

**Comment by KrzysztofCwalina** [comment]
@dotnet-bot test Innerloop Windows_NT Release Build and Test

**Comment by KrzysztofCwalina** [comment]
@dotnet-bot test Innerloop Windows_NT Debug Build and Test

**Comment by KrzysztofCwalina** [comment]
@dotnet-bot test Innerloop Windows_NT Release Build and Test

**Pull request merged**